restcomm/restcomm.mscontrol.jsr309/src/main/java/org/restcomm/connect/mscontrol/jsr309/Jsr309CallController.java

```diff
@@ -831,6 +832,7 @@ private void onStop(Stop message, ActorRef self, ActorRef sender) {
                 this.mediaGroup.getSignalDetector().stop();
                 this.collecting = Boolean.FALSE;
             }
+            call.tell(new RecordStoped(), self);
```

Comment:
do we need to provide smae change for non JSR309 controller?

Reply:
@jaimecasero 
Line 791:
https://github.com/RestComm/Restcomm-Connect/pull/2956/files/2772b7dc7d20cc069d1ae12ebb06e39f55019371#diff-57215364d1e805510727c150e410959bL791

This line of code hold the thread for more than 2 seconds every time RestComm want to stop recording from XMS.
When finish Record verb RC does:
 VoiceInterpreter send signal to stop recording to Jsr309CallController actor.
Jsr309CallController thread was block at line 791 for 2 second.
In the mean time Jsr309CallController is blocked, VoiceInterpreter send signal to close media session, but Jsr309CallController is blocked so the signal was ignored and then BYE cannot send to XMS.

In non JSR309, the code does not have any block point, so DELETE MGCP still send to RMS.